src/file-utils/lms_file_utils/directory_repository.py

```diff
@@ -0,0 +1,197 @@
+    base_directory: str, section_id: int
+) -> Optional[str]:
+    """
+    Gets the canonical directory for section associations files.
+
+    Parameters
+    ----------
+    base_directory: str
+        The base / parent directory for LMS extractor files.
+    section_id: int
+        Section identifier.
+
+    Returns
+    -------
+    str or None
+        Directory as string or None if the section associations directory does not exist.
+    """
+    return _get_directory_for_section(
+        base_directory, section_id, "section-associations"
+    )
+
+
+def get_section_activities_directory(
+    base_directory: str, section_id: int
+) -> Optional[str]:
+    """
+    Gets the canonical directory for section activities files.
+
+    Parameters
+    ----------
+    base_directory: str
+        The base / parent directory for LMS extractor files.
+    section_id: int
+        Section identifier.
+
+    Returns
+    -------
+    str or None
+        Directory as string or None if the section activities directory does not exist.
+    """
+    return _get_directory_for_section(base_directory, section_id, "section-activities")
+
+
+def get_assignments_directory(base_directory: str, section_id: int) -> Optional[str]:
+    """
+    Gets the canonical directory for assignments files.
+
+    Parameters
+    ----------
+    base_directory: str
+        The base / parent directory for LMS extractor files.
+    section_id: int
+        Section identifier.
+
+    Returns
+    -------
+    str or None
+        Directory as string or None if the assignments directory does not exist.
+    """
+    return _get_directory_for_section(base_directory, section_id, "assignments")
+
+
+def get_grades_directory(base_directory: str, section_id: int) -> Optional[str]:
+    """
+    Gets the canonical directory for grades files.
+
+    Parameters
+    ----------
+    base_directory: str
+        The base / parent directory for LMS extractor files.
+    section_id: int
+        Section identifier.
+
+    Returns
+    -------
+    str or None
+        Directory as string or None if the grades directory does not exist.
+    """
+    return _get_directory_for_section(base_directory, section_id, "grades")
+
+
+def get_submissions_directory(
+    base_directory: str, section_id: int, assignment_id: int
+) -> Optional[str]:
+    """
+    Gets the canonical directory for submissions files.
+
+    Parameters
+    ----------
+    base_directory: str
+        The base / parent directory for LMS extractor files.
+    section_id: int
+        Section identifier.
+    assignment_id: int
+        Assignment identifier.
+
+    Returns
+    -------
+    str or None
+        Directory as string or None if the submissions directory does not exist.
+    """
+    assignments = _get_directory_for_section(
+        base_directory, section_id, f"assignment={assignment_id}"
+    )
+
+    if assignments is None:
+        return None
+
+    return os.path.join(assignments, "submissions")

```

Comment:
I'd move strings to the top as constants and then call them when needed.
`SUBMISSIONS = "submissions"`
```suggestion
    return os.path.join(assignments, SUBMISSIONS)

```

Reply:
Same for all strings

Reply:
done